sympy/physics/mechanics/joint.py

```diff
@@ -543,3 +543,137 @@ def _set_linear_velocity(self):
         self.child_point.set_pos(self.parent_point, 0)
         self.child.masscenter.v2pt_theory(self.parent.masscenter,
                                           self.parent.frame, self.child.frame)
+
+
+class SlidingJoint(Joint):
+    """Sliding (Prismatic) Joint.
+
+    Explanation
+    ===========
+
+    It is defined such that the child body translates with respect to the parent
+    body along the body fixed parent axis. The point of joint (sliding joint) is defined
+    by two points in each body which coincides initially.
+
+    Parameters
+    ==========
+
+    name : string
+        The Joint's name which makes it unique.
+    parent : Body
+        The parent body of joint.
+    child : Body
+        The child body of joint.
+    coordinates: dynamicsymbol, optional
+        Coordinates of joint.
+    speeds : dynamicsymbol, optional
+        Speeds of joint.
+    parent_joint_pos : Vector, optional
+        Defines the joint's point where the parent will be connected to child.
+        Default value is masscenter of Parent Body.
+    child_joint_pos : Vector, optional
+        Defines the joint's point where the child will be connected to parent.
+        Default value is masscenter of Child Body.
+    parent_axis : Vector, optional
+        Axis of parent frame which would be be aligned with child's
+        axis. Default is x axis in parent's frame.
+    child_axis : Vector, optional
+        Axis of child frame which would be aligned with parent's
+        axis. Default is x axis in child's frame.
+
+    Attributes
+    ==========
+
+    name : string
+        The joint's name.
+    parent : Body
+        The joint's parent body.
+    child : Body
+        The joint's child body.
+    coordinate : dynamicsymbol, optional
+        Coordinate of the joint.
+    speed : dynamicsymbol, optional
+        Speed of the joint.
+    parent_joint : Point
+        The joint's point where parent body is connected to the joint.
+    child_joint : Point
+        The joint's point where child body is connected to the joint.
+    parent_axis : Vector
+        The axis of parent frame.
+    child_axis : Vector
+        The axis of child frame.
```

Comment:
I updated the wording for all these parameters and attributes in the pin joint pr. Make them consistent here.